### PR TITLE
make `IRShow.method_name` inferrable

### PR DESCRIPTION
**Pull request by KristofferC**
4 methods pushes it over the limit. 

Seems to have the same effect as https://github.com/JuliaLang/julia/pull/49605.

Now:

```
julia> code_warntype(Base.IRShow.method_name, Tuple{Core.LineInfoNode})

MethodInstance for Base.IRShow.method_name(::Core.LineInfoNode)
  from method_name(m::Core.LineInfoNode) @ Base.IRShow compiler/ssair/show.jl:185
Arguments
  #self#::Core.Const(Base.IRShow.method_name)
  m::Core.LineInfoNode
Body::Symbol
1 ─      nothing
│   %2 = Base.getproperty(m, :method)::Any
│   %3 = Base.IRShow.normalize_method_name(%2)::Symbol
└──      return %3
```

### CODE REVIEW

**Review by vtjnash — COMMENTED** [review]
This seems somewhat less good code structure. But we also should be using === to compare Symbols not ==

**Comment by KristofferC** [comment]
> This seems somewhat less good code structure.

Yes and if the results where equivalent I would have kept the previous one.

File: base/stacktraces.jl [hunk 1]
```diff
@@ -134,7 +134,7 @@ function lookup_inline_frame_info(func::Symbol, file::Symbol, linenum::Int, inli
             # backtrack to find the matching MethodInstance, if possible
             for j in (i - 1):-1:1
                 nextline = inlinetable[j]
-                nextline.inlined_at == line.inlined_at && Base.IRShow.method_name(line) == Base.IRShow.method_name(nextline) && line.file == nextline.file || break
+                nextline.inlined_at == line.inlined_at && Base.IRShow.method_name(line) === Base.IRShow.method_name(nextline) && line.file == nextline.file || break
```
Comment (vtjnash) [inline base/stacktraces.jl:137]: I think `file` is also required (inferably) to be a Symbol?

**Comment by vtjnash** [comment]
> > This seems somewhat less good code structure.
> 
> Yes and if the results where equivalent I would have kept the previous one.

Touché. Does the `===` change fix this and/or should we add a type assert for Symbol in the call wrapper of this that seems may be present already?

**Comment by KristofferC** [comment]
The issue is already fixed by making the function inferable, but for consistency it is indeed good to fix so that all symbol comparisons are made with `===`

**Review by timholy — APPROVED** [review]

**Pull request merged**